File: src/main/java/uk/co/threebugs/mochiwhattotrade3/setupgroup/SetupGroup.java
package uk.co.threebugs.mochiwhattotrade3.setupgroup;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import uk.co.threebugs.mochiwhattotrade3.setupgroups.SetupGroups;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Entity
@Table(name = "setup_group")
public final class SetupGroup {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id;

    @ManyToOne(fetch = FetchType.EAGER)
    @JoinColumn(name = "setup_groups_id")
    private SetupGroups setupGroups;

    private String path;
    private String symbol;
    private Boolean enabled;
}

File: src/main/kotlin/uk/co/threebugs/mochiwhattotrade3/setupgroup/SetupGroup.kt
package uk.co.threebugs.mochiwhattotrade3.setupgroup

import jakarta.persistence.*
import uk.co.threebugs.mochiwhattotrade3.setupgroups.SetupGroups

@Entity
@Table(name = "setup_group")
class SetupGroup (
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    var id: Int? = null,

    @ManyToOne(fetch = FetchType.EAGER)
    @JoinColumn(name = "setup_groups_id")
    var setupGroups: SetupGroups? = null,
    var path: String? = null,
    var symbol: String? = null,
    var enabled: Boolean? = null,
){
}
